test/torchaudio_unittest/models/wav2vec2/huggingface_intergration_test.py

```diff
@@ -159,6 +164,50 @@ def test_import_finetune(self, config, _):
         self._test_import_pretrain(original.wav2vec2, imported, config)
         self._test_import_finetune(original, imported, config)
 
+    @WAVLM_CONFIGS
+    def test_import_pretrain_wavlm(self, config, _):
+        """WavLM models from HF transformers can be imported and yield the same results"""
+        original = self._get_model(config).eval()
+        imported = import_huggingface_model(original).eval()
+
+        # FeatureExtractor
+        x = torch.randn(3, 1024)
+        ref = original.feature_extractor(x).transpose(1, 2)
+        hyp, _ = imported.feature_extractor(x, None)
+        self.assertEqual(ref, hyp)
+        # Feature projection
+        x = torch.randn(3, 10, config["conv_dim"][-1])
+        ref = original.feature_projection(x)[0]
+        hyp = imported.encoder.feature_projection(x)
+        self.assertEqual(ref, hyp)
+        # Convolutional Positional Encoder
+        x = torch.randn(3, 256, config["hidden_size"])
+        ref = original.encoder.pos_conv_embed(x)
+        hyp = imported.encoder.transformer.pos_conv_embed(x)
+        self.assertEqual(ref, hyp)
+
+        position_bias = None
+        position_bias_imp = None
+        for original_, imported_ in zip(original.encoder.layers, imported.encoder.transformer.layers):
```

Comment:
Could we check that both `original.encoder.layers` and `imported.encoder.transformer.layers` have non-zero values and they are the same length?

This can be simply achieved with [`strict` option in Python 3.10](https://peps.python.org/pep-0618/), but until we drop the support for 3.9, this needs to be done manually.

Reply:
or you can introduce [`zip_common` like in torchtext](https://github.com/pytorch/text/blob/main/test/torchtext_unittest/common/case_utils.py#L45-L54) (and replace all the `zip` in test as a separate PR (different bootcamp task))